### PR TITLE
add 'est-total' statistic token

### PR DESCRIPTION
**Pull request by bastibuck**
Hey,

as proposed in https://github.com/fabiospampinato/vscode-todo-plus/issues/253 this adds a new statistic token `estTotal`.

While adding the new capability I also refactored the `add()` to make it more readable.

Please give feedback and let me know what you think.

Cheers

### CODE REVIEW

**Comment by fabiospampinato** [comment]
It looks good overall, but can you:

- Rename `estTotal` to `est-total`.
- Rename `Total estimation` to `Total estimated time`
- Revert the beautification, let's keep this PR's diff shorter, and I don't actually remember why I wrote that section like that, maybe it's performance critical or something.
- I think you should list `est-total` here too and in general it would be wise to search repo-wise for `'est'`:

https://github.com/fabiospampinato/vscode-todo-plus/blob/eb2df3a13af438a50ef46e7fb5f8661da5542596/src/utils/statistics.ts#L132

**Comment by bastibuck** [comment]
Sure did.

Not sure why you wrote the now un-beautified part like you did ;-) 
It works though so I reverted this part and only added the `est-total` code

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Manage todo lists with ease. Powerful, easy to use and customizable. [View the d
 - **[TaskPaper](https://www.taskpaper.com) compatible**: just set `todo.symbols.box`, `todo.symbols.done` and `todo.symbols.cancelled` to `-`
 - **Timekeeping**: you can mark todos as started and track elapsed time until completion
 - **Timer**: a timer can be displayed in the statusbar for started todos
-- **Time estimates**: you can estimate the time it will take to complete a todo by adding a tag to it that looks like this: `@est(3 hours)`, `@est(2h30m)` or `@2h30m`. Then you can use the `[est]` token in statistics
+- **Time estimates**: you can estimate the time it will take to complete a todo by adding a tag to it that looks like this: `@est(3 hours)`, `@est(2h30m)` or `@2h30m`. Then you can use the `[est]` and `[est-total]` token in statistics
```
Comment (fabiospampinato) [inline README.md:25]: It should be "token**s**".

File: README.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ It adds 6 shortcuts when editing a `Todo` file:
   "todo.timekeeping.finished.format": "YY-MM-DD HH:mm", // Format used for displaying time inside @done/cancelled
   "todo.timekeeping.elapsed.enabled": true, // Enable the @lasted/wasted tag
   "todo.timekeeping.elapsed.format": "short-compact", // Format used for displaying time diff inside @lasted/waster
-  "todo.timekeeping.estimate.format": "short-compact", // Format used for the `[est]` token
+  "todo.timekeeping.estimate.format": "short-compact", // Format used for the `[est]` and `[est-total]` token
```
Comment (fabiospampinato) [inline README.md:111]: "token**s**" here too.

File: src/utils/statistics.ts [hunk 3]
```diff
@@ -30,12 +30,17 @@ const Statistics = {
...
 
-      } else if ( !disabledTokens.est && !disabledEst && ( prefix === 'e' || ( prefix >= '0' && prefix <= '9' ) ) ) { // Maybe @est(1h20m) or @1h20m
+      } else if ( !disabledTokens.est && ( prefix === 'e' || ( prefix >= '0' && prefix <= '9' ) ) ) { // Maybe @est(1h20m) or @1h20m
 
-        tokens.estSeconds += Statistics.timeTags.parseEstimate ( tag );
+        // always add to total
+        tokens.estTotalSeconds += Statistics.timeTags.parseEstimate ( tag );
 
-      }
+        // only add if not disabled
```
Comment (fabiospampinato) [inline src/utils/statistics.ts:38]: These comments are necessary I think.

File: src/utils/statistics.ts [hunk 4]
```diff
@@ -30,12 +30,17 @@ const Statistics = {
...
+        if(!disabledEst) {
```
Comment (fabiospampinato) [inline src/utils/statistics.ts:39]: I'd write it like so: `if ( !disabledEst ) {`

File: README.md (hunk 2 above)
Comment (fabiospampinato) [reply README.md:111]: Also you should update the list of settings in the readme too.

File: README.md (hunk 1 above)
Comment (bastibuck) [reply README.md:25]: Nice catch! Will update those the upcoming days :) Looking forward to using this

File: README.md (hunk 2 above)
Comment (bastibuck) [reply README.md]: > Also you should update the list of settings in the readme too.

Not sure what you mean 🤔

File: src/utils/statistics.ts (hunk 4 above)
Comment (bastibuck) [reply src/utils/statistics.ts]: See newest commit

File: src/utils/statistics.ts (hunk 3 above)
Comment (bastibuck) [reply src/utils/statistics.ts]: See newest commit

File: README.md (hunk 1 above)
Comment (bastibuck) [reply README.md]: See newest commit

**Comment by fabiospampinato** [comment]
Thanks, published in v4.17.

**Pull request merged**